agent/asm/asm_test.go

```diff
@@ -109,3 +109,34 @@ func TestASMGetAuthConfig(t *testing.T) {
 		})
 	}
 }
+
+func TestGetSecretFromASM(t *testing.T) {
+	cases := []struct {
+		Name        string
+		Resp        secretsmanager.GetSecretValueOutput
+		ShouldError bool
+	}{
+		{
+			Name: "SuccessWithValidResponse",
```

Comment:
should be testing other cases here?

Reply:
For secret, there is no cases like wrong json struct or missing attribute which is true in asm auth. So no other cases need to be tested.